t/Privileges.t

```diff
@@ -0,0 +1,35 @@
+use strict;
+use warnings;
+
+use Test;
+use Win32;
+
+plan tests => 6;
+
+ok(ref(Win32::GetProcessPrivileges) eq 'HASH');
+ok(ref(Win32::GetProcessPrivileges(Win32::GetCurrentProcessId())) eq 'HASH');
+
+# All Windows PIDs are divisible by 4. It's an undocumented implementation
+# detail, but it means it's extremely unlikely that the PID below is valid.
+ok(!Win32::GetProcessPrivileges(3423237));
+
+my $whoami = `whoami /priv 2>&1`;
+my $skip = ($? == -1 || $? >> 8) ? '"whoami" command is missing' : 0;
+
+skip($skip, sub{
+    my $privs = Win32::GetProcessPrivileges();
+
+    while ($whoami =~ /^(Se\w+)/mg) {
+        return 0 unless exists $privs->{$1};
+    }
+
+    return 1;
+});
+
+# there isn't really anything to test, we just want to make sure that the
+# function doesn't segfault
```

Comment:
If `Win32::IsDeveloperModeEnabled` returns `false`, you could verify that `Win32::IsSymlinkCreationAllowed` returns `false` as well, but maybe that is not very interesting.

But if `Win32::IsSymlinkCreationAllowed` returns `true` you should check that you can actually create a symlink.  (and if it is false, check that symlink creation fails).

Reply:
That makes sense. That will require File::Temp, but that shouldn't be problem, it was core since 5.6.1.